itest/litd_custom_channels_test.go

```diff
@@ -3652,3 +3652,114 @@ func testCustomChannelsForwardBandwidth(ctxb context.Context,
+	oracleAddr := fmt.Sprintf("localhost:%d", port.NextAvailablePort())
+	oracle := newOracleHarness(oracleAddr)
+	oracle.start(t.t)
+	t.t.Cleanup(oracle.stop)
+
+	ctxb := context.Background()
+	lndArgs := slices.Clone(lndArgsTemplate)
+	litdArgs := slices.Clone(litdArgsTemplateNoOracle)
+	litdArgs = append(litdArgs, fmt.Sprintf(
+		"--taproot-assets.experimental.rfq.priceoracleaddress="+
+			"rfqrpc://%s", oracleAddr,
+	))
+
+	// For this test, Zane will be our dedicated Universe server for all parties.
+	zane, err := net.NewNode(
+		t.t, "Zane", lndArgs, false, true, litdArgs...,
+	)
+	require.NoError(t.t, err)
+
+	litdArgs = append(litdArgs, fmt.Sprintf(
+		"--taproot-assets.proofcourieraddr=%s://%s",
+		proof.UniverseRpcCourierType, zane.Cfg.LitAddr(),
+	))
+
+	// We'll just make a single node here, as this doesn't actually rely on a set
+	// of active channels.
+	alice, err := net.NewNode(t.t, "Alice", lndArgs, false, true, litdArgs...)
+	aliceTap := newTapClient(t.t, alice)
+
+	// Next, we'll make a new asset with a specified decimal display. We'll also
+	// make grouped asset as well.
+	usdMetaData := &taprpc.AssetMeta{
+		Data: []byte(`{
+"description":"this is a USD stablecoin with decimal display of 6"
+}`),
+		Type: taprpc.AssetMetaType_META_TYPE_JSON,
+	}
+
+	const decimalDisplay = 6
+	itestAsset = &mintrpc.MintAsset{
+		AssetType: taprpc.AssetType_NORMAL,
+		Name:      "USD",
+		AssetMeta: usdMetaData,
+		// We mint 1 million USD with a decimal display of 6, which
+		// results in 1 trillion asset units.
+		Amount:          1_000_000_000_000,
+		DecimalDisplay:  decimalDisplay,
+		NewGroupedAsset: true,
+	}
+
+	// Mint an asset on Charlie and sync Dave to Charlie as the universe.
+	mintedAssets := itest.MintAssetsConfirmBatch(
+		t.t, t.lndHarness.Miner.Client, aliceTap,
+		[]*mintrpc.MintAssetRequest{
+			{
+				Asset: itestAsset,
+			},
+		},
+	)
+	usdAsset := mintedAssets[0]
+	assetID := usdAsset.AssetGenesis.AssetId
+
+	// Now that we've minted the asset, we can set the price in the oracle.
+	var id asset.ID
+	copy(id[:], assetID)
+
+	// We'll assume a price of $100,000.00 USD for a single BTC. This is just the
+	// current subjective price our oracle will use. From this BTC price, we'll
+	// scale things up to be in the precision of the asset we minted above.
+	btcPrice := rfqmath.NewBigIntFixedPoint(
+		100_000_000, 2,
+	).ScaleTo(decimalDisplay)
+	oracle.setPrice(id, btcPrice, btcPrice)
+
+	// Now we'll make a normal invoice for 1 BTC using Alice.
+	expirySeconds := 10
+	amountSat := 100_000_000
+	invoiceResp, err := alice.AddInvoice(ctxb, &lnrpc.Invoice{
+		Value:  int64(amountSat),
+		Memo:   "normal invoice",
+		Expiry: int64(expirySeconds),
+	})
+	require.NoError(t.t, err)
+
+	payReq := invoiceResp.PaymentRequest
+
+	// Now that we have our payment request, we'll call into the new decode asset
+	// pay req call.
+	decodeResp, err := aliceTap.DecodeAssetInvoice(ctxb, &taprpc.DecodeAssetPayReq{
```

Comment:
The `AddAssetSellOrder` RPC has a (somewhat hidden) `SkipAssetChannelCheck` flag that circumvents the requirement to have an actual asset channel (just any channel with the peer is enough). We could also add that flag to the `DecodeAssetInvoice` RPC (could make it hidden or just not add it on the CLI at all), allowing this to be an itest in `tapd`.

Reply:
> allowing this to be an itest in tapd

I think even with the version as is, it can be a test in `tapd` given it just hits the oracle directly. 

Assuming we add the `AddAssetSellOrder` variant, then how could it be a test in tapd, if it still needs the `lnd` peer to be connected?